src/LNLib/Geometry/Curve/NurbsCurve.cpp

```diff
@@ -3578,27 +3605,26 @@ std::vector<double> LNLib::NurbsCurve::GetParamsOnCurve(const LN_NurbsCurve& cur
 std::vector<LNLib::XYZ> LNLib::NurbsCurve::Tessellate(const LN_NurbsCurve& curve)
 {
 	int degree = curve.Degree;
-	std::vector<double> knotVector = curve.KnotVector;
-	std::vector<XYZW> controlPoints = curve.ControlPoints;
+	const std::vector<double>& knotVector = curve.KnotVector;
+	const std::vector<XYZW>& controlPoints = curve.ControlPoints;
 
 	if (degree == 1)
 	{
 		return ControlPointsUtils::ToXYZ(controlPoints);
 	}
 
-	double start = knotVector[0];
-	double end = knotVector[knotVector.size() - 1];
-
-	XYZ startPoint = controlPoints[0].ToXYZ(true);
-	XYZ endPoint = controlPoints[controlPoints.size() - 1].ToXYZ(true);
-
+	std::vector<double> unqiueKnotVector = knotVector;
+	unqiueKnotVector.erase(std::unique(unqiueKnotVector.begin(), unqiueKnotVector.end()), unqiueKnotVector.end());
```

Comment:
The accuracy of double (Approximately 1E-15) may cause the std:: unique function to fail in removing duplicate nodes. 
You can skip processing intervals with a length of 0 in the for loop below.

Reply:
Alternatively, a double comparator can be added to std::unique.

Reply:
OK!  when two double parameter are close, I use 'continue'.